src/Aspire.Hosting/ApplicationModel/EndpointReference.cs

```diff
@@ -55,6 +55,7 @@ internal string GetExpression(EndpointProperty property = EndpointProperty.Url)
             EndpointProperty.Port => "port",
             EndpointProperty.Scheme => "scheme",
             EndpointProperty.TargetPort => "targetPort",
+            EndpointProperty.ContainerPort => "containerPort",
```

Comment:
We don't want a new expression here, azd will have to change to understand this.